Tests/AsyncKitTests/ConnectionPoolTests.swift

```diff
@@ -134,6 +134,30 @@ final class ConnectionPoolTests: XCTestCase {
             // pass
         }
     }
+    
+    // https://github.com/vapor/async-kit/issues/63
+    func testDeadlock() {
+        let foo = FooDatabase()
+        let pool = EventLoopConnectionPool(
+            source: foo,
+            maxConnections: 1,
+            creationTimeout: .seconds(1),
```

Comment:
Could this be reduced to e.g. ~100ms to improve test runtimes? Or, even better, is there an event loop group for testings that lets us "speed up time" for testing?